litellm/proxy/db/db_spend_update_writer.py

```diff
@@ -0,0 +1,692 @@
+                                        team_id, response_cost
+                                    )
+                                )
+                                batcher.litellm_teamtable.update_many(  # 'update_many' prevents error from being raised if no row exists
+                                    where={"team_id": team_id},
+                                    data={"spend": {"increment": response_cost}},
+                                )
+                    prisma_client.team_list_transactions = (
+                        {}
+                    )  # Clear the remaining transactions after processing all batches in the loop.
+                    break
+                except DB_CONNECTION_ERROR_TYPES as e:
+                    if (
+                        i >= n_retry_times
+                    ):  # If we've reached the maximum number of retries
+                        _raise_failed_update_spend_exception(
+                            e=e,
+                            start_time=start_time,
+                            proxy_logging_obj=proxy_logging_obj,
+                        )
+                    # Optionally, sleep for a bit before retrying
+                    await asyncio.sleep(2**i)  # Exponential backoff
+                except Exception as e:
+                    _raise_failed_update_spend_exception(
+                        e=e, start_time=start_time, proxy_logging_obj=proxy_logging_obj
+                    )
+
+        ### UPDATE TEAM Membership TABLE with spend ###
+        team_member_list_transactions = db_spend_update_transactions[
+            "team_member_list_transactions"
+        ]
+        if (
+            team_member_list_transactions is not None
+            and len(team_member_list_transactions.keys()) > 0
+        ):
+            for i in range(n_retry_times + 1):
+                start_time = time.time()
+                try:
+                    async with prisma_client.db.tx(
+                        timeout=timedelta(seconds=60)
+                    ) as transaction:
+                        async with transaction.batch_() as batcher:
+                            for (
+                                key,
+                                response_cost,
+                            ) in team_member_list_transactions.items():
+                                # key is "team_id::<value>::user_id::<value>"
+                                team_id = key.split("::")[1]
+                                user_id = key.split("::")[3]
+
+                                batcher.litellm_teammembership.update_many(  # 'update_many' prevents error from being raised if no row exists
+                                    where={"team_id": team_id, "user_id": user_id},
+                                    data={"spend": {"increment": response_cost}},
+                                )
+                    prisma_client.team_member_list_transactions = (
```

Comment:
it's confusing that this will update prisma_client.<in_memory_transactions> but it's reading 

```
team_member_list_transactions = db_spend_update_transactions[
            "team_member_list_transactions"
        ]
```